### PR TITLE
Rename the `rust` manifest key to `edition`

### PR DESCRIPTION
**Pull request by alexcrichton**
This'll hopefully jive better with the terminology of "edition" throughout the
rest of Rust!

### CODE REVIEW

**Comment by alexcrichton** [comment]
There's some discussion at https://gitter.im/rust-lang/WG-epoch?at=5aeb2d9653ceca3604ac3bae and https://paper.dropbox.com/doc/Edition-preview-cheat-sheet-HhfGRmTX4mByEOWWI2P7o, but I think the general consensus is towards `edition` rather than `rust`

**Comment by matklad** [comment]
Great!

@bors r+

File: src/doc/src/reference/unstable.md [hunk 1]
```diff
@@ -190,7 +190,7 @@ cargo-features = ["edition"]
 
 [package]
 ...
-rust = "2018"
+edition = "2018"
```
Comment (ehuss) [inline src/doc/src/reference/unstable.md]: It should probably say `edition key` (instead of `rust`) a couple lines above this.

Comment (matklad) [reply src/doc/src/reference/unstable.md]: Excellent catch!

**Pull request merged**